src/driver_camd_midi.c

```diff
@@ -0,0 +1,410 @@
+    if (GetVar("DefMidiOut", _outport, sizeof(_outport), 0)) {
+      retname = _outport;
+    }
+
+    UnlockCAMD(key);
+  }
+
+  return retname;
+}
+
+static int InitCamd(void) {
+
+  if (!(CamdBase = OpenLibrary("camd.library", 0))) {
+    g_error = CamdDrv_FailedOpenCamdLibrary;
+    goto failure;
+  }
+  if (!(RealTimeBase = OpenLibrary("realtime.library", 0))) {
+    g_error = CamdDrv_FailedOpenRealTimeLibrary;
+    goto failure;
+  }
+
+  g_midiNode = CreateMidi(MIDI_MsgQueue, 0L, MIDI_SysExSize, 0, MIDI_Name,
+                          (Tag) "SW Midi Out", TAG_END);
+  if (!g_midiNode) {
+    g_error = CamdDrv_Error;
+    goto failure;
+  }
+
+  const char *deviceName = FindMidiDevice();
+  if (!deviceName) {
+    g_error = CamdDrv_Error;
+    goto failure;
+  }
+
+  g_midiLink = AddMidiLink(g_midiNode, MLTYPE_Sender, MLINK_Location,
+                           (Tag)deviceName /*"out.0"*/, TAG_END);
+  if (!g_midiLink) {
+    g_error = CamdDrv_Error;
+    goto failure;
+  }
+
+  return CamdDrv_Success;
+
+failure:
+  ShutDownCamd();
+  return g_error;
+}
+int CamdDrv_GetError() { return 0; }
+
+const char *CamdDrv_ErrorString(int ErrorNumber) {
+  switch (ErrorNumber) {
+  case CamdDrv_Success:
+    return "CamdDrv_Success";
+  case CamdDrv_Error:
+    return "CamdDrv_Error";
+  case CamdDrv_FailedOpenCamdLibrary:
+    return "CamdDrv: failed to open camd.library";
+  case CamdDrv_FailedOpenRealTimeLibrary:
+    return "CamdDrv: failed to open realtime.library";
+  default:
+    return "unknown error";
+  }
+}
+
+int CamdDrv_MIDI_Init(midifuncs *funcs, const char *params) {
+
+  g_error = CamdDrv_Success;
+
+  if (InitCamd() != CamdDrv_Success) {
+    return MUSIC_Error;
+  }
+
+  memset(funcs, 0, sizeof(midifuncs));
+
+  funcs->NoteOff = Func_NoteOff;
+  funcs->NoteOn = Func_NoteOn;
+  funcs->PolyAftertouch = Func_PolyAftertouch;
+  funcs->ControlChange = Func_ControlChange;
+  funcs->ProgramChange = Func_ProgramChange;
+  funcs->ChannelAftertouch = Func_ChannelAftertouch;
+  funcs->PitchBend = Func_PitchBend;
+  funcs->SysEx = Func_SysEx;
+
+  // FIXME: what about these? None of the other midi driver implements them
+  //  void ( *ReleasePatches )( void );
+  //  void ( *LoadPatch )( int number );
+  //  void ( *SetVolume )( int volume );
+  //  int  ( *GetVolume )( void );
+
+  return MUSIC_Ok;
+}
+
+void CamdDrv_MIDI_Shutdown() { ShutDownCamd(); }
+
+int CamdDrv_MIDI_StartPlayback(void (*service)()) {
+
+  g_service = service;
+
+  if (!g_serviceTask) {
+    g_MainTask = FindTask(NULL);
+
+    BPTR file = 0;
+    //    Open("PROGDIR:log.txt", MODE_READWRITE);
+    //    Seek(file, 0, OFFSET_END);
+    //    FPrintf(file, "Creating new Task  -------------------\n");
+
+    // Set priority to 21, so it is just a bit higher than input and mouse
+    // movements won't slow down playback
+    if (!(g_serviceTask = (struct Task *)CreateNewProcTags(
+              NP_Name, (Tag) "SW MidiService", NP_Priority, 21, NP_Entry,
```

Comment:
21 was chosen empirically. But yes, 25 would also work.
I have  yet to try the whole thing on a real machine... the slowdown will probably not happen there. I believe when it happens my PC is just not fast enough for the emulation since everything (gameplay and music) slows down at the same pace.